WordPress/Classes/System/WordPressAppDelegate+openURL.swift

```diff
@@ -0,0 +1,132 @@
+                                                             sourceApplication: options[UIApplication.OpenURLOptionsKey.sourceApplication] as? String,
+                                                             annotation: options[UIApplication.OpenURLOptionsKey.annotation]) {
+            returnValue = true
+        }
+
+        // 3. let's see if it's our wpcom scheme
+        guard url.scheme == WPComScheme else {
+            return returnValue
+        }
+
+        // this works with our custom ~=
+        switch url {
+        case "magic-login":
+            returnValue = handleMagicLogin(url: url)
+        case "viewpost":
+            returnValue = handleViewPost(url: url)
+        case "viewstats":
+            returnValue = handleViewStats(url: url)
+        case "debugging":
+            handleDebugging(url: url)
+        default:
+            break
+        }
+
+        return returnValue
+    }
+
+    private func handleMagicLogin(url: URL) -> Bool {
+        DDLogInfo("App launched with authentication link")
+        let allowWordPressComAuth = !AccountHelper.isDotcomAvailable()
+        guard let rvc = window.rootViewController else {
+            return false
+        }
+        return WordPressAuthenticator.openAuthenticationURL(url,
+                                                            allowWordPressComAuth: allowWordPressComAuth,
+                                                            fromRootViewController: rvc)
+    }
+
+    private func handleViewPost(url: URL) -> Bool {
+        guard let params = url.queryParams(),
+            let blogId = params.intValue(of: "blogId"),
+            let postId = params.intValue(of: "postId") else {
+            return false
+        }
+
+        WPTabBarController.sharedInstance()?.showReaderTab(forPost: NSNumber(value: postId), onBlog: NSNumber(value: blogId))
+
+        return true
+    }
+
+    private func handleViewStats(url: URL) -> Bool {
+        let blogService = BlogService(managedObjectContext: ContextManager.sharedInstance().mainContext)
+
+        guard let params = url.queryParams(),
+            let siteId = params.intValue(of: "siteId"),
+            let blog = blogService.blog(byBlogId: NSNumber(value: siteId)) else {
+            return false
+        }
+
+        let statsViewController = StatsViewController()
+        statsViewController.blog = blog
+        statsViewController.dismissBlock = {
+            WPTabBarController.sharedInstance()?.dismiss(animated: true, completion: nil)
+        }
+
+        let navController = UINavigationController(rootViewController: statsViewController)
+        navController.modalPresentationStyle = .currentContext
+        navController.navigationBar.isTranslucent = false
+
+        WPTabBarController.sharedInstance()?.present(navController, animated: true, completion: nil)
+
+        return true
+    }
+
+    @nonobjc private func handleDebugging(url: URL) {
```

Comment:
Is there a specific reason this is `nonobjc`?

Reply:
AH, good catch. I initially had this method returning an enum to deal with the fact that it doesn't affect `returnValue`, then I decided if it's crashing then it doesn't really matter 😜